packages/labs/redux/src/lib/connector.ts

```diff
@@ -0,0 +1,138 @@
+   *
+   * If none is provided, the controller will not subscribe to Redux store
+   * changes nor provide any selected value. Do this if you only wish to bring
+   * in the `dispatch` method to the component.
+   */
+  selector?: Selector<ReturnType<S['getState']>, V>;
+  /**
+   * Function used to check whether a selected value is different from
+   * previously selected value.
+   *
+   * Defaults to triple equals which will suffice for directly selecting values
+   * out of the state that's updating with immutable pattern, or if using a
+   * memoized selector using a library like `reselect`.
+   *
+   * Provide a custom function here if the selector returns derived data
+   * that's not memoized.
+   */
+  equalityCheck?: EqualityCheck;
+};
+
+/**
+ * Reactive controller which subscribes to a Redux store and ties changes to a
+ * custom element's reactive life cycle.
+ *
+ * This controller must be used with a provider that supplies the Redux store
+ * using the `storeContext` from this package using the [Context
+ * Protocol](https://github.com/webcomponents-cg/community-protocols/blob/main/proposals/context.md).
+ */
+export class Connector<S extends Store, V> implements ReactiveController {
+  private _host: ReactiveControllerHost & HTMLElement;
+  private _store!: S;
+  private _selector: ConnectorOptions<S, V>['selector'];
+  private _equalityCheck: EqualityCheck;
+  private _unsubscribe?: () => void;
+  private _selected?: V;
+
+  /**
+   * Static method to create a typed version of the `Connector` constructor.
+   * This allows type checking for the `selector` option, the selected value, as
+   * well as the `dispatch` method provided by the connector.
+   *
+   * @example
+   *
+   * ```ts
+   * // `AppStore` type gotten from created Redux store
+   * export const AppConnector = Connector.withStoreType<AppStore>();
+   *
+   * // Usage within component – `state` will already be typed
+   * new AppConnector(this, {selector: (state) => state.counter.value});
+   * ```
+   */
+  static withStoreType<S extends Store>(): new <V>(
+    host: ReactiveControllerHost & HTMLElement,
+    options?: ConnectorOptions<S, V>
+  ) => Connector<S, V> {
+    return this;
+  }
+
+  /**
+   * Selected value.
+   */
+  get selected() {
+    return this._selected as V;
+  }
+
+  /**
+   * Method to dispatch an action to the store.
+   */
+  get dispatch(): S['dispatch'] {
+    return (action) => this._store.dispatch(action);
+  }
+
+  constructor(
+    host: ReactiveControllerHost & HTMLElement,
+    options?: ConnectorOptions<S, V>
+  ) {
+    this._host = (host.addController(this), host);
+    this._selector = options?.selector;
+    this._equalityCheck = options?.equalityCheck ?? tripleEquals;
+  }
+
+  hostConnected() {
+    new ContextConsumer(this._host, {
+      context: storeContext,
+      subscribe: true,
+      callback: (store) => {
+        this._store = store as S;
+      },
+    });
+    if (this._store === undefined) {
```

Comment:
Since this context subscribes, is this error correct here? What if the page is using a ContextRoot? Maybe this should be some kind of debug warning.

Reply:
I had that thought too..

I was hoping if it might be possible to kind of force this requirement on users to force the context provider be always set up first because I otherwise we have to consider that without having the store on start up, the `selected` value could be undefined and users need to always guard for that.

So if a user selected for an object, they would basically have to always null check any property access to it for their first render. Maybe that's not the worst thing, but I really like the idea that, assuming the user provided the correct store type and a valid selector, they can always expect a certain value to exist and use as such within their component code.